### ISSUE TITLE
Compile *.cpp as *.mm

### ISSUE THREAD

**Issue by Wizermil** [facebook/buck#1454]
I'm working on a large mobile cross platform project and often you can see in Xcode a couple of cpp files compiled as objective-c++. This feature allows our code to be shared among many other platforms. Unfortunately that doesn't translate well to BUCK.

I tried to change the compile flag of the src, but they are appended not overwritten so it doesn't work
```
srcs = [
    ('sample.cpp', ['-x objective-c++']),
],
```

Any ideas to make it work?

**Comment by njlr** [comment]
Perhaps you could pull the Objective C++ files out into their own target and set the flags in `compiler_flags`? 

```
cxx_library(
)
```

**Comment by yiding** [comment]
per-file flags should come at the end of the command line so it can overwrite earlier flags. The fact that it isn't doing that is a bug and we should fix it.

**Comment by yiding** [comment]
Fixed in 5f06d90f3aa9ca5830e4746e5f1515182eb68760.
